test/gtest/uct/test_sockaddr.cc

```diff
@@ -1043,6 +1050,69 @@ UCS_TEST_P(test_uct_sockaddr, ep_disconnect_err_codes)
 UCT_INSTANTIATE_SOCKADDR_TEST_CASE(test_uct_sockaddr)
 
 
+class test_uct_sockaddr_listener : public test_uct_sockaddr {
+protected:
+    static void conn_request_cb(
+            uct_listener_h listener, void *arg,
+            const uct_cm_listener_conn_request_args_t *conn_req_args)
+    {
+        test_uct_sockaddr_listener *self =
+                static_cast<test_uct_sockaddr_listener*>(arg);
+
+        ASSERT_FALSE(ucs_test_all_flags(
+                self->m_state,
+                (TEST_STATE_SERVER_LISTENER_READY_CLOSE |
+                 TEST_STATE_SERVER_LISTENER_CLOSED)));
+
+        self->handle_conn_request(listener, conn_req_args,
+                                  test_uct_sockaddr::server_connect_cb,
+                                  test_uct_sockaddr::server_disconnect_cb);
+
+        // Wait until listener will be set for being destroyed
+        while (!(self->m_state & TEST_STATE_SERVER_LISTENER_READY_CLOSE)) {
+            std::this_thread::yield();
+        }
+
+        // Wait to increase the probability of trying to lock mutex during
+        // destroying while it is already acquired by this thread
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
I hope for smth better

Reply:
currently, we don't have a good solution to guarantee 100% reproducibility